cfme/ansible_tower/jobs.py

```diff
@@ -51,14 +61,79 @@ def is_displayed(self):
         )
 
 
+class AnsibleTowerJobsDetailsView(TowerJobsView):
+    @View.nested
+    class entities(View):  # noqa
+        """ Represents details page when it's switched to Summary/Table view """
+        properties = SummaryTable(title="Properties")
+        relationships = SummaryTable(title="Relationships")
+        smart_management = SummaryTable(title="Smart Management")
+
+    @property
+    def is_displayed(self):
+        """Is this view being displayed?"""
+        return self.title.text == '{} (Summary)'.format(self.context['object'].template_name)
+
+
 @attr.s
-class TowerJobs(BaseEntity):
-    pass
+class TowerJob(BaseEntity):
+    template_name = attr.ib()
+
+    @property
+    def status(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of("Status")
+
+    def delete(self):
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove this Job', handle_alert=True)
+        view.flash.assert_no_error()
+
+    def is_job_successful(self):
+        return True if self.status == "successful" else False
+
+    def wait_for_completion(self, num_sec=1200, delay=10):
+        def last_status():
+            logger.info("Last status message in UI: '{}'".format(self.status))
+
+        wait_for(self.is_job_successful, num_sec=num_sec, delay=delay, fail_func=last_status,
+                 message="Job finished")
 
 
 @attr.s
 class TowerJobsCollection(BaseCollection):
-    ENTITY = TowerJobs
+    ENTITY = TowerJob
+
+    def all(self):
+        view = navigate_to(self, 'All')
+        return [
+            self.instantiate(template_name=e.data["template_name"])
+            for e in view.entities.get_all(surf_pages=True)
+        ]
+
+    def status(self, template_name):
+        """Get status of a specific job from the All page."""
+        view = navigate_to(self, 'All')
+        for e in view.entities.get_all(surf_pages=True):
+            if e.data["template_name"] == template_name:
+                return e.data["status"]
+        raise NoSuchElementException("No job named {}".format(template_name))
+
+    def delete_all(self):
+        view = navigate_to(self, 'All')
+        view.paginator.check_all()
+        view.configuration.item_select('Remove Jobs', handle_alert=True)
+        view.flash.assert_no_error()
+
+    def is_job_successful(self, template_name):
+        try:
+            return True if self.status(template_name) == "successful" else False
+        except NoSuchElementException:
+            return None
+
+    @property
+    def are_all_jobs_successful(self):
+        return all([self.is_job_successful(job.template_name) for job in self.all()])
```

Comment:
Again I'm questioning the implementation and use of such a convenience method on the collection.

This isn't taking advantage of the fact that the status can be read from the collection page - its actually instantiating entities and using their status properties, navigating to every entity in turn.  I think its very misleading as a collection method.

I don't see its use anywhere here, and am doubting that its necessary for test automation.